packages/sanity/src/_internal/cli/actions/backup/chooseBackupIdPrompt.ts

```diff
@@ -0,0 +1,41 @@
+import type {CliCommandContext} from '@sanity/cli'
+import {defaultApiVersion} from '../../commands/backup/backupGroup'
+import resolveApiClient from './resolveApiClient'
+
+// maxBackupIdsShown is the maximum number of backup IDs to show in the prompt.
+// Higher numbers will cause the prompt to be slow.
+const maxBackupIdsShown = 100
+
+async function chooseBackupIdPrompt(
+  context: CliCommandContext,
+  datasetName: string,
+): Promise<string> {
+  const {prompt} = context
+
+  const {projectId, token, client} = await resolveApiClient(context, datasetName, defaultApiVersion)
+
+  // Fetch last $maxBackupIdsShown backups for this dataset.
+  // We expect here that API returns backups sorted by creation date in descending order.
+  const response = await client.request({
+    headers: {Authorization: `Bearer ${token}`},
+    uri: `/projects/${projectId}/datasets/${datasetName}/backups`,
+    query: {limit: maxBackupIdsShown.toString()},
+  })
+
+  if (response && response.backups && response.backups.length > 0) {
```

Comment:
💡 could this simply be the following suggestion?

```suggestion
  if (response?.backups?.length > 0) {
```

Reply:
Nice. Simplified it.